src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/ValidateResourceTypeFilterAttribute.cs

```diff
@@ -24,13 +24,18 @@ public override void OnActionExecuting(ActionExecutingContext context)
             if (context.RouteData.Values.TryGetValue(KnownActionParameterNames.ResourceType, out var actionModelType) &&
                 context.ActionArguments.TryGetValue(KnownActionParameterNames.Resource, out var parsedModel))
             {
-                if (!string.Equals((string)actionModelType, ((Base)parsedModel).TypeName, StringComparison.OrdinalIgnoreCase))
-                {
-                    throw new ResourceNotValidException(new List<ValidationFailure>
+                ValidateType((Resource)parsedModel, (string)actionModelType);
+            }
+        }
+
+        private static void ValidateType(Resource resource, string expectedType)
```

Comment:
Since this is no longer being used outside of this filter, can we make it private to this class (or revert to the way it was before)?